pennylane/qchem/vibrational/vibrational_class.py

```diff
@@ -0,0 +1,126 @@
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains functions and classes to generate a pes object.
+This object stores all the necessary information to construct
+vibrational Hamiltonian for a given molecule."""
+
+import numpy as np
+
+import pennylane as qml
+
+from ..openfermion_pyscf import _import_pyscf
+
+# pylint: disable=import-outside-toplevel, unused-variable
+
+BOHR_TO_ANG = 0.529177
+
+
+def harmonic_analysis(scf_result, method="rhf"):
+    r"""Performs harmonic analysis by evaluating the Hessian using PySCF routines.
+
+    Args:
+       scf_result: pyscf object from electronic structure calculations
+       method: Electronic structure method to define the level of theory
+            for harmonic analysis. Default is restricted Hartree-Fock 'rhf'.
+
+    Returns:
+       pyscf object containing information about the harmonic analysis
+    """
+    pyscf = _import_pyscf()
+    from pyscf.hessian import thermo
+
+    method = method.strip().lower()
+    if method not in ["rhf", "uhf"]:
+        raise ValueError(f"Specified electronic structure method, {method} is not available.")
+
+    hess = getattr(pyscf.hessian, method).Hessian(scf_result).kernel()
+    harmonic_res = thermo.harmonic_analysis(scf_result.mol, hess)
+    return harmonic_res
+
+
+def single_point(molecule, method="rhf"):
+    r"""Runs electronic structure calculation.
+
+    Args:
+      molecule: Molecule object.
+      method: Electronic structure method to define the level of theory.
+              Default is restricted Hartree-Fock 'rhf'.
+
+    Returns:
+      pyscf object from electronic structure calculation
+    """
+    pyscf = _import_pyscf()
+
+    method = method.strip().lower()
+    if method not in ["rhf", "uhf"]:
+        raise ValueError(f"Specified electronic structure method, {method}, is not available.")
+
+    geom = [
+        [symbol, tuple(np.array(molecule.coordinates)[i])]
+        for i, symbol in enumerate(molecule.symbols)
+    ]
+    spin = int((molecule.mult - 1) / 2)
+    mol = pyscf.gto.Mole(atom=geom, symmetry="C1", spin=spin, charge=molecule.charge, unit="Bohr")
+    mol.basis = molecule.basis_name
+    mol.build()
+    if method == "rhf":
+        scf_obj = pyscf.scf.RHF(mol).run(verbose=0)
+    else:
+        scf_obj = pyscf.scf.UHF(mol).run(verbose=0)
+    return scf_obj
+
+
+def _import_geometric():
+    """Import geometric."""
+    try:
+        import geometric
+    except ImportError as Error:
+        raise ImportError(
+            "This feature requires geometric. It can be installed with: pip install geometric."
+        ) from Error
+
+    return geometric
+
+
+def optimize_geometry(molecule, method="rhf"):
```

Comment:
No need to be user-facing, if it is called by other functions.

Reply:
We discussed having these functions user facing, so that a user can generate potential energy surfaces without having to go through the wrapper function for vibrational class.